cluster/service.go

```diff
@@ -46,16 +48,33 @@ type Service struct {
 // nodes.
 // Raft store will be initialized and ready to be started. To start the service call Open().
 func New(selector cluster.NodeSelector, cfg Config) *Service {
-	addr := fmt.Sprintf("%s:%d", cfg.Host, cfg.RPCPort)
+	raftListenAddress := fmt.Sprintf("%s:%d", cfg.Host, cfg.RPCPort)
+	// When using FQDN lookup we might want to advertise a different IP than the one we'll be listening on.
+	// This address is then sent to raft peers as the address this node listen on.
+	// This address needs to proxy/forward to that node (think static ip for a service)
+	// This is necessary to ensure that the FQDN ip will be stored in the raft logs.
+	raftAdvertisedAddress := raftListenAddress
```

Comment:
**thought:** 
won't be `= raftListenAddress` if `cfg.EnableFQDNResolver` not enabled 

which will change port from `cfg.RaftPort` to `cfg.RPCPort` ? 

do you think we shall make it `fmt.Sprintf("%s:%d", cfg.Host, cfg.RaftPort)` ?

Reply:
You are completely correct thanks for catching it. I'm wondering why all our tests passed however 🤔 . I think because we splice out the port we communicate in RAFT and re-use the `raftport` that is configured on the remote node it didn't matter.

More port tech debt and assumption that are wrong 👎